src/usage/search.md

```diff
@@ -3,38 +3,81 @@
 
-* Less than `64` rows should be returned. It's expected.
-* The vector index returned `64` rows, but `32` of which are deleted before but the index do not know since PostgreSQL vacuum is lazy.
-* The vector index returned `64` rows, but `32` of which are invisible to the transaction so PostgreSQL decided to hide these rows for you.
-* The vector index returned `64` rows, but `32` of which are satisfying the condition `id % 2 = 0` in `WHERE` clause.
+There are two search modes: `basic` and `vbase`.
 
-There are four ways to solve the problem:
+### `basic`
 
-* Set `vectors.k` larger. If you estimate that 20% of rows will satisfy the condition in `WHERE`, just set `vectors.k` to be 5 times than before.
-* Set `vectors.enable_vector_index` to `off`. If you estimate that 0.0001% of rows will satisfy the condition in `WHERE`, just do not use vector index. No algorithms will be faster than brute force by PostgreSQL.
-* Set `vectors.enable_prefilter` to `on`. If you cannot estimate how many rows will satisfy the condition in `WHERE`, leave the job for the index. The index will check if the returned row can be accepted by PostgreSQL. However, it will make queries slower so the default value for this option is `off`.
-* Set `vectors.enable_vbase` to `on`. It will use vbase optimization, so that the index will pull rows as many as you need. It only works for HNSW algorithm.
+`basic` is the default search mode. In this mode, vector indexes behave like a vector search library. It works well if all of your queries is like this:
 
-:::
+```sql
+SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' LIMIT 5;
+```
+
+It's recommended if your do not take advantages of
+
+* database transaction
+* deletions without `VACUUM`
+* WHERE clauses and very complex SQL statments
+
+### `vbase`
+
+`basic` is another search mode. In this mode, vector indexes behave like a database index.
+
+Assuming you are using HNSW algorithm, you may want the following SQL to work:
+
+```sql
+SET vectors.hnsw_ef_search = 64;
+SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' WHERE id % 2 = 0 LIMIT 64;
+```
+
+In `search` mode, you may only get `32` rows because the HNSW algorithm does search simply so the filter condition is ignored.
+
+In `vbase` mode, the HNSW algorithm is guaranteed to return rows as many as you need, so you can always get correct behavior if your do take advantages of:
+
+* database transaction
+* deletions without `VACUUM`
+* `WHERE` clauses and very complex SQL statments
+
+You can enable `vbase` by a SQL statment `SET vectors.search_mode = vbase;`.
+
+## Prefilter
+
+If your queries include a `WHERE` clause, you can set set search mode to `vbase`. It's good and it even works on all conditions. `vbase` is a **postfilter** method: it pulls rows as many as you need, but it scans rows that you may not need. Since some rows will definitely be removed by the `WHERE` clause, we can skip scanning them, which will make the search faster. We call it **prefilter**.
```

Comment:
1. redudant set
2. Personally, the definition of `vbase` is better to be at `vbase` chapter
> `vbase` is a **postfilter** method: it pulls rows as many as you need, but it scans rows that you may not need.

Reply:
We provide a view of vector search library versus database vector index in search mode section. And here we provide a view of prefiltering versus postfilter. I think it's OKay.